substrate/client/service/src/config.rs

```diff
@@ -323,3 +287,64 @@ impl From<PathBuf> for BasePath {
 		BasePath::new(path)
 	}
 }
+
+/// RPC configuration.
+#[derive(Debug)]
+pub struct RpcConfiguration {
```

Comment:
Nice refactor indeed, but with the introduced `addr: Vec<RpcEndpoint>` all the configurations/settings is already included in there and these other fields are not used by the server at all.

So, it should be possible to remove this struct an just use `Option<Vec<RpcEndpoint>` IIRC.

Reply:
Ok, I was wrong it can be none